From 6e2513d4e155393f75cc6a5b58c9a678718b1b84 Mon Sep 17 00:00:00 2001
From: Song.Li@windriver.com
Date: Mon, 27 Aug 2012 16:45:06 +0800
Subject: [PATCH] Disable the x86 optimized assembly code

In sysdeps/gcc/x86.h file,the assembly code in function
AO_compare_double_and_swap_double_full conflict with
-fno-omit-frame-poniter gcc option,
-fno-omit-frame-poniter is useful for oprofile in wrlinux,
so disable the optimized assembly code.

Upstream-Status:Inappropriate
---
 src/atomic_ops/sysdeps/gcc/x86.h |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/src/atomic_ops/sysdeps/gcc/x86.h b/src/atomic_ops/sysdeps/gcc/x86.h
index df02a46..a64f45f 100644
--- a/src/atomic_ops/sysdeps/gcc/x86.h
+++ b/src/atomic_ops/sysdeps/gcc/x86.h
@@ -129,6 +129,10 @@ AO_compare_and_swap_full(volatile AO_t *addr, AO_t old, AO_t new_val)
 }
 #define AO_HAVE_compare_and_swap_full
 
+#if 0
+/*This function conflict with -fno-omit-frame-pointer gcc option
+ *Disable it */
+
 /* Returns nonzero if the comparison succeeded. */
 /* Really requires at least a Pentium.          */
 AO_INLINE int
@@ -164,5 +168,6 @@ AO_compare_double_and_swap_double_full(volatile AO_double_t *addr,
   return (int) result;
 }
 #define AO_HAVE_compare_double_and_swap_double_full
+#endif
 
 #include "../ao_t_is_int.h"
-- 
1.7.7

